lintrunner_adapters/adapters/docformatter_linter.py

```diff
@@ -0,0 +1,105 @@
+# SPDX-FileCopyrightText: Copyright 2025 Arm Limited and/or its affiliates <open-source-office@arm.com>
+# SPDX-License-Identifier: MIT
+"""Adapter for https://github.com/PyCQA/docformatter"""
+
+import argparse
+import shutil
+import tempfile
+from pathlib import Path
+from typing import Optional
+
+from lintrunner_adapters import (
+    LintMessage,
+    LintSeverity,
+    add_default_options,
+    run_command,
+)
+
+LINTER_CODE = "DOCFORMATTER"
+
+
+def check_file(
+    filename: str,
+    retries: int,
+    config: str,
+) -> list[LintMessage]:
+    tmp_path: Optional[Path] = None
+
+    try:
+        path = Path(filename)
+        original = path.read_text(encoding="utf-8")
+
+        # Docformatter cannot print the would-be results to stdout, so
+        # formatting must be applied in-place to an actual file. A temp file is
+        # used for this.
+        with tempfile.NamedTemporaryFile(suffix=".py", delete=False) as tmp_file:
+            tmp_path = Path(tmp_file.name)
+        shutil.copyfile(path, tmp_path)
+
+        args = ["docformatter"]
+        if config is not None:
+            args += ["--config", config]
+        args += ["--in-place", str(tmp_path)]
```

Comment:
Is it possible to not use the `--in-place` option and instead apply the printed diff, according to https://docformatter.readthedocs.io/en/latest/usage.html#use-from-the-command-line ?

Reply:
The printed diff to stdout is a unified diff. If I understand Lintrunner correctly, that won't result in proper original/replacement parameters. That's why I used in-place on a temporary file and use that as the "replacement" parameter. It's no in place on the actual file.

Reply:
Is it possible to use some python code to apply the diff on the original content? difflib?

Reply:
I suppose so. We could use the patch tool and run that with subprocess and print it to stdout and capture it in the "replacement" variable. I don't know if this works on other platforms than Linux though. The current approach should work for other OS as well.

What I could find from difflib is used for creating unified diffs. 

Is it the temp file you'd like to avoid?

Reply:
Done. Let me know what you think